flixel/input/actions/FlxActionSet.hx

```diff
@@ -0,0 +1,322 @@
+		analogActions.push(Action);
+		return true;
+	}
+	
+	public function destroy():Void
+	{
+		FlxDestroyUtil.destroyArray(digitalActions);
+		FlxDestroyUtil.destroyArray(analogActions);
+		digitalActions = null;
+		analogActions = null;
+	}
+	
+	/**
+	 * Remove a digital action from this set
+	 * @param	Action a FlxActionDigital
+	 * @param	Destroy whether to destroy it as well
+	 * @return	whether it was found and removed
+	 */
+	public function removeDigital(Action:FlxActionDigital, Destroy:Bool = true):Bool
+	{
+		var index = digitalActions.indexOf(Action);
+		if (index != -1)
+		{
+			if (Destroy)
+				digitalActions[index].destroy();
+			digitalActions.splice(index, 1);
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 * Remove an analog action from this set
+	 * @param	Action a FlxActionAnalog
+	 * @param	Destroy whether to destroy it as well
+	 * @return	whether it was found and removed
+	 */
+	public function removeAnalog(Action:FlxActionAnalog, Destroy:Bool = true):Bool
+	{
+		var index = analogActions.indexOf(Action);
+		if (index != -1)
+		{
+			if (Destroy)
+				analogActions[index].destroy();
+			analogActions.splice(index, 1);
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 * Update all the actions in this set (each will check inputs & potentially trigger)
+	 */
+	public function update():Void
+	{
+		if (!active) return;
+		for (digitalAction in digitalActions)
+		{
+			digitalAction.update();
+		}
+		for (analogAction in analogActions)
+		{
+			analogAction.update();
+		}
+	}
+	
+	private function attachSteamControllerSub(Handle:Int, Attach:Bool, InputType:FlxInputType, DigitalActions:Array<FlxActionDigital>, AnalogActions:Array<FlxActionAnalog>)
+	{
+		var length = InputType == FlxInputType.DIGITAL ? DigitalActions.length : AnalogActions.length;
+		
+		for (i in 0...length) 
+		{
+			var action = InputType == FlxInputType.DIGITAL ? DigitalActions[i] : AnalogActions[i];
+			
+			if (action.steamHandle != -1)	//all steam-affiliated actions will have this numeric ID assigned
+			{
+				var inputExists = false;
+				var theInput:FlxActionInput = null;
+				
+				//check if any of the steam controller inputs match this handle
+				for (input in action.inputs)
+				{
+					if (input.device == FlxInputDevice.STEAM_CONTROLLER && input.deviceID == Handle)
+					{
+						inputExists = true;
+						theInput = input;
+					}
+				}
+				
+				if (Attach)
+				{
+					//attaching: add inputs for this controller if they don't exist
+					
+					if (!inputExists)
+					{
+						if (InputType == FlxInputType.DIGITAL)
+						{
+							DigitalActions[i].addInput(new FlxActionInputDigitalSteam(action.steamHandle, FlxInputState.JUST_PRESSED, Handle));
+						}
+						else if (InputType == FlxInputType.ANALOG)
+						{
+							AnalogActions[i].addInput(new FlxActionInputAnalogSteam(action.steamHandle, FlxAnalogState.MOVED, FlxAnalogAxis.EITHER, Handle));
+						}
+					}
+				}
+				else
```

Comment:
you can change following lines to:
```
else if (inputExists)
{
    action.removeInput(theInput);
}
```
to avoid nested conditional (unless you plan to add some other logic there)